File: onDeck/Utilities/StreamLinkRouter.swift
import Foundation

enum StreamLinkRouter {
    /// Routes a broadcast callSign to the appropriate streaming platform URL.
    static func url(for game: Game) -> URL {
        let exclusiveBroadcast = game.broadcasts.first { $0.isExclusive }

        guard let callSign = exclusiveBroadcast?.callSign else {
            return mlbTVURL(gamePk: game.id)
        }

        switch callSign {
        case "Peacock":
            return URL(string: "https://www.peacocktv.com/sports/mlb")!
        case "Apple TV", "Apple TV+":
            return URL(string: "https://tv.apple.com/us/room/edt.item.62327df1-6e37-4222-86c1-056489e15668")!
        case "ESPN", "ESPN2":
            return URL(string: "https://www.espn.com/watch/")!
        case "Netflix":
            return URL(string: "https://www.netflix.com")!
        case "TBS":
            return URL(string: "https://www.tbs.com/mlb-on-tbs")!
        default:
            return mlbTVURL(gamePk: game.id)
        }
    }

    private static func mlbTVURL(gamePk: Int) -> URL {
        URL(string: "https://www.mlb.com/tv/g\(gamePk)")!
    }
}

File: windows/src/OnDeck.Core/Utilities/StreamLinkRouter.cs
using OnDeck.Core.Models;

namespace OnDeck.Core.Utilities;

/// <summary>Port of <c>Utilities/StreamLinkRouter.swift</c>.</summary>
public static class StreamLinkRouter
{
    /// <summary>Routes a broadcast callSign to the appropriate streaming platform URL.</summary>
    public static Uri Url(Game game)
    {
        var callSign = game.Broadcasts.FirstOrDefault(broadcast => broadcast.IsExclusive)?.CallSign;

        return callSign switch
        {
            "Peacock" => new Uri("https://www.peacocktv.com/sports/mlb"),
            "Apple TV" or "Apple TV+" =>
                new Uri("https://tv.apple.com/us/room/edt.item.62327df1-6e37-4222-86c1-056489e15668"),
            "ESPN" or "ESPN2" => new Uri("https://www.espn.com/watch/"),
            "Netflix" => new Uri("https://www.netflix.com"),
            "TBS" => new Uri("https://www.tbs.com/mlb-on-tbs"),
            _ => MlbTvUrl(game.Id),
        };
    }

    private static Uri MlbTvUrl(int gamePk) => new($"https://www.mlb.com/tv/g{gamePk}");
}
